service/storage_service.cc

```diff
@@ -682,6 +682,31 @@ bool storage_service::do_handle_cdc_generation(db_clock::time_point ts) {
     }).get0();
 }
 
+namespace {
+class cdc_generation_handling_nonfatal_exception : public std::runtime_error {
+    using std::runtime_error::runtime_error;
+};
+
+constexpr char could_not_retrieve_msg_template[]
+        = "Could not retrieve CDC streams with timestamp {} upon gossip event. Reason: \"{}\". Action: {}.";
```

Comment:
Wouldn't string_view be better? I'm not an expert but my impression was that it's a recommended way to use string literals.

Reply:
Tried googling for arguments pro/con and found nothing conclusive. I can switch to the `""sv` literal for the sole sake of writing "modern C++" :)

Reply:
I meant sth like this https://github.com/scylladb/scylla/blob/9d9d54c804dd400d90edfdb84c8f5319d06b1633/gms/feature_service.cc#L30

Reply:
Tried it - apparently our `fmt::format` is not ready for `string_view`:
```
service/storage_service.cc:779:23: note:   cannot convert ‘service::{anonymous}::could_not_retrieve_msg_template’ (type ‘const string_view’ {aka ‘const std::basic_string_view<char>’}) to type ‘const char*’
  779 |         cdc_log.error(could_not_retrieve_msg_template, ts, std::current_exception(), "not retrying");
```

Reply:
Ok. Let's leave that then.